ecs-init/docker/docker.go

```diff
@@ -215,6 +216,30 @@ func (c *Client) StartAgent() (int, error) {
 	return c.docker.WaitContainer(container.ID)
 }
 
+// GetContainerLogTail will return the last logWindowSize lines of logs for
+// the Agent Container.
+func (c *Client) GetContainerLogTail(logWindowSize string) string {
+	containerToLog, _ := c.findAgentContainer()
+	if containerToLog == "" {
+		log.Info("No existing container to take logs from.")
+	}
+	// we want to capture some logs from our removed containers in case of failure
+	var containerLogBuf bytes.Buffer
+	err := c.docker.Logs(godocker.LogsOptions{
```

Comment:
This API requires a Docker log driver that supports reading logs.  Not all of them do.  Even though we don't explicitly specify the log driver when starting the agent container, customers can set the log driver at the daemon level.  You'll need to expect failures to happen in cases where customers have configured a different default log driver.

Reply:
Agree.  I believe this is a good starting point for the set of cases without a different default log driver.  Presently we have no visibility into these failed (and reaped) agent containers.

Reply:
> This API requires a Docker log driver that supports reading logs. Not all of them do. 

This will work in the default case, and fail gracefully other cases. It's not perfect, but it still means we capture failures in a few more failure scenarios.

IMO the long term fix would be to have agent no longer manage its own log file. That change is much more invasive to existing setups, however.